src/game/connect4.rs

```diff
@@ -65,20 +89,281 @@ impl GameAdapter for Connect4Adapter<'_> {
+            while (lenr < CONNECT_FOUR)
+                && (col_aux < COL_SIZE)
+                && (self.get_cell_at(row_aux, col_aux) == Some(self.turn))
+            {
+                lenl += 1;
+                col_aux += 1;
+            }
+            if lenl + lenr >= CONNECT_FOUR {
+                return true;
+            }
+
+            //3.) LeftUp + RightDown
+            lenl = 0;
+            lenr = 0;
+            row_aux = row;
+            col_aux = column;
+            ok = true;
+            while (lenl < CONNECT_FOUR)
+                && (ok)
+                && (row_aux < ROW_SIZE)
+                && (self.get_cell_at(row_aux, col_aux) == Some(self.turn))
+            {
+                lenl += 1;
+                if col_aux > 0 {
+                    col_aux -= 1;
+                } else {
+                    ok = false;
+                }
+                row_aux += 1;
+            }
+            ok = true;
+            while (lenr < CONNECT_FOUR)
+                && (col_aux < COL_SIZE)
+                && (ok)
+                && (self.get_cell_at(row_aux, col_aux) == Some(self.turn))
+            {
+                lenl += 1;
+                if row_aux > 0 {
+                    row_aux -= 1;
+                } else {
+                    ok = false;
+                }
+                col_aux += 1;
+            }
+            if lenl + lenr >= CONNECT_FOUR {
+                return true;
+            }
+
+            //4.) LeftDomn + RightUp
+            lenl = 0;
+            lenr = 0;
+            row_aux = row;
+            col_aux = column;
+            ok = true;
+            while (lenl < CONNECT_FOUR)
+                && (ok)
+                && (self.get_cell_at(row_aux, col_aux) == Some(self.turn))
+            {
+                lenl += 1;
+                if row_aux > 0 {
+                    row_aux -= 1;
+                } else {
+                    ok = false;
+                }
+                if col_aux > 0 {
+                    col_aux -= 1;
+                } else {
+                    ok = false;
+                }
+            }
+            while (lenr < CONNECT_FOUR)
+                && (col_aux < COL_SIZE)
+                && (row_aux < ROW_SIZE)
+                && (self.get_cell_at(row_aux, col_aux) == Some(self.turn))
+            {
+                lenl += 1;
+                col_aux += 1;
+                row_aux += 1;
+            }
+            lenl + lenr >= CONNECT_FOUR
+        } else {
+            false
+        }
+    }
+    fn draw_move(&self) -> bool {
+        self.board.iter().all(|ref col| col.len() == ROW_SIZE)
+    }
+
+    fn moves(&mut self, column: usize) -> actix_web::Result<()> {
+        if self.completed {
+            return Err(actix_web::Error::from(GameAdapterError::InvalidGameState(
+                State::Ended,
+            )));
+        }
+        self.insert_move_if_legal(column)?;
+        Ok(())
+    }
+    fn visualize(&mut self, column: usize) {
```

Comment:
This is not used outside of testing so it should have the `#[cfg(test)]` attribute